models/DataObject/ClassDefinition/Data/Relations/AbstractRelations.php

```diff
@@ -179,12 +179,8 @@ public function load(Localizedfield|AbstractData|\Pimcore\Model\DataObject\Objec
 
         // using PHP sorting to order the relations, because "ORDER BY index ASC" in the queries above will cause a
         // filesort in MySQL which is extremely slow especially when there are millions of relations in the database
-        usort($relations, function ($a, $b) {
-            if ($a['index'] == $b['index']) {
-                return 0;
-            }
-
-            return ($a['index'] < $b['index']) ? -1 : 1;
+        usort($relations, static function ($a, $b) {
```

Comment:
Should we add type hints?
```suggestion
        usort($relations, static function (array $a, array $b): int {
```

Reply:
Funnily enough adding type declarations makes performance degrade since it adds a type check on every item in the array. Are we sure we want it where it is known and performance is important?

Reply:
I would always go for better performance. Do you have a rough estimation of the degradation? How much slower are we talking?

Reply:
About 2-4%, so most of the times the benefits type hints outweigh the cost.

25k rows on M1 Mac with PHP 8.1.16, time in seconds.

<table class="generateTable">
  <thead>
    <tr><th>No type hints</th>
    <th>With argument and return type hints</th>
  </tr>
</thead>
<tr>
    <td>0.513512134552</td>
    <td>0.53477501869202</td>
</table>

Reply:
Nice, thank you for the info. Let's take an average of 3% then, which can have a big impact. 
So yes, let's keep it like in your last commit.